src/Api/RestClient.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KafkaRest\Api;
+
+use KafkaRest\Api\Value\Request\ConsumerOptions;
+use KafkaRest\Api\Value\Request\Subscription;
+use KafkaRest\Api\Value\Response\AssignedPartition;
+use KafkaRest\Api\Value\Response\Consumer;
+use KafkaRest\Api\Value\Response\Message;
+use KafkaRest\Api\Value\Response\Offset;
+use KafkaRest\Api\Value\Response\Partition;
+use KafkaRest\Api\Value\Response\ProduceResult;
+use KafkaRest\Api\Value\Response\Topic;
+
+interface RestClient
+{
+    /**
+     * GET /topics
+     *
+     * @return array<string>
+     */
+    public function listTopics() : array;
+
+    /**
+     * GET /topics/(string:topic_name)
+     */
+    public function getTopic(string $topic) : Topic;
+
+    /**
+     * POST /topics/(string:topic_name)
+     *
+     * @param Value\Request\Message[] $messages
+     *
+     * @return ProduceResult[]
```

Comment:
why not use `array<T>` syntax everywhere? already used in `listTopics`

Reply:
I had it everywhere and then removed it. The reason is simple: Phpstorm doesn't understand it, not even in the new EAP. And I don't really want to use fancy new syntax just because it's there, even though only half of the tooling I use supports it (and the lesser-used half to top that off). I use the syntax where I don't mind that Phpstorm doesn't understand it, like with strings in `listTopics`. Using double annotations works, but just for return types, so it's only a partial solution. I decided to skip the syntax until it's fully supported. This works for now, for both Phpstorm and PHPStan.